### ISSUE TITLE
Cannot invoke "org.apache.lucene.index.FieldInfo.getVectorDimension()" because "info" is null, and how to fix?

### ISSUE THREAD

**Issue by hetao29** [elastic/elasticsearch#114468]
### Elasticsearch Version

8.15.1

### Java Version

_bundled_

### OS Version

docker

### Problem Description

GET /item/_stats?error_trace=true

```json
{
  "_shards": {
    "total": 2,
    "successful": 0,
    "failed": 2,
    "failures": [
      {
        "shard": 0,
        "index": "item_1683611971",
        "status": "INTERNAL_SERVER_ERROR",
        "reason": {
          "type": "null_pointer_exception",
          "reason": """Cannot invoke "org.apache.lucene.index.FieldInfo.getVectorDimension()" because "info" is null"""
        }
      }
    ]
  },
  "_all": {
    "primaries": {},
    "total": {}
  },
  "indices": {}
}
```

### Steps to Reproduce

add vector field

### 
#112720 fixed?

**Comment by AI-IshanBhatt** [comment]
@hetao29 As per #112720, it is fixed but it requires version 8.15.2 and above. 
Can you update ES version and try again ?

**Comment by hetao29** [comment]
> [@hetao29](https://github.com/hetao29) As per [#112720](https://github.com/elastic/elasticsearch/pull/112720), it is fixed but it requires version 8.15.2 and above. Can you update ES version and try again ?

I upgraded, and It's ok, thanks.

**Comment by AI-IshanBhatt** [comment]
@hetao29 I am closing this issue in that case. Feel free to contact if the issue resurfaces.
